File: src/2020/3/python/3.2.py
with open("src/2020/3/inputs.txt", "r") as f:
    inputs = list(f.read().splitlines())
    slopes = [
        (1, 1),
        (3, 1),
        (5, 1),
        (7, 1),
        (1, 2)
    ]
    width = len(inputs[0])
    answer = 1
    for slopeX, slopeY in slopes:
        counterY = 0
        counter = 0
        for line in inputs:
            if line[(counterY * slopeX) % width] == "#":
                counter += 1
            counterY += 1
        answer *= counter
    print(answer)

File: src/2020/3/js/3.2.js
fs = require("fs");
fs.readFile("src/2020/3/inputs.txt", "utf8", function (err, data) {
  let inputs = data.split("\n");
  console.time();
  const width = inputs[0].length;
  const slopes = [
    [1, 1],
    [3, 1],
    [5, 1],
    [7, 1],
    [1, 2],
  ];
  let answer = 1;
  slopes.forEach(([slopeX, slopeY]) => {
    let counter = 0;
    let counter2 = 0;
    for (let i = 0; i < inputs.length; i += slopeY) {
      counter += inputs[i][(counter2 * slopeX) % width] === "#" ? 1 : 0;
      counter2++;
    }
    console.log(counter);
    answer *= counter;
  });
  console.log(answer);
  console.timeEnd();
});
